Framework/Core/include/Framework/runDataProcessing.h

```diff
@@ -124,6 +129,16 @@ int main(int argc, char** argv)
     LOG(ERROR) << "Unknown error while setting up workflow.";
   }
 
+  char* idstring = nullptr;
+  for (int argi = 0; argi < argc; argi++) {
```

Comment:
Is `--id` a new option needed for this hook? I think not?
In that case, perhaps the id parameter is already available from somewhere else, and does not need to be parsed from the cmdline again?

Reply:
The `--id` parameter is added by the main driver so that every child process knows it's role in the workflow. So it is there already. The parsing of all the options is done in `doMain` in the cxx file. The customization happens in the header file because of some template trick, so one would need to pass the function as another parameter to do main.

Using this id parameter allows to customize the hook depending on the position of a process in the workflow.

I decided not to do this and implement some simple parsing. But well, that can be changed.